Survival-DEV/blog-backend
# Blog 37: Refactor the code | PART II

### :dart:  What is been done so far:
1. [x] Add the Gurads(`JwtAuthGuard`) to `POST, PUT, DELETE` http Methods of all endpoints
2. [x] Rename users, blogs, tags, categories, comments services
3. [x] Remove unnecessary  packages, such as:
   - `rxjs`
4. [x] Update the paths @ tsConfig.json & update the import based on it
   **Related to #37**

----
### :compass:  Testing
Steps to test my PR:
1. `npm ci`
2. make sure to setup your `env` file to have these params:
   <details>
      <summary>Expand to see the params</summary>
    
      > DATABASE_URL=<your db url#>
      > NODE_ENV=DEBUG  //for now
      > logDB=["query", "error"]
      > PORT=<your port#>
      > TRUSTED_IP=["127.0.0.1"]
      > WHITELIST=['http://localhost']
      > ALLOWED_METHODS='GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS'
      > JWT_VERIFICATION_TOKEN_SECRET='secreto2'
      > JWT_VERIFICATION_TOKEN_EXPIRATION_TIME='60s'
      > EMAIL_CONFIRMATION_URL='http://localhost:5000/api/confirm-email/'
      > SEND_GRID_ACCESS_KEY=`<your sendgird key>`
      > FROM_EMAIL=`your sendgrid email`
      > SENDGRID_TEMPLATE_ID=d-3285e7571ee64e8fafedb163996e5a55
      >CONTACT_EMAIL= `<some_mail for_swagger_configs>`
   </details>
4. `docker-compose up`   --to fire redis 
3. `npm run start:dev`   --in another terminal
